rooms.js

```diff
@@ -163,6 +165,107 @@ var Room = (function () {
 		}
 		return true;
 	};
+	//mute handling
+	Room.prototype.runMuteTimer = function () {
+		if (this.muteTimer || this.mutedUsers.length === 0) return;
+		var timeUntilExpire = this.mutedUsers[0].time - Date.now();
+		if (timeUntilExpire <= 0) {
+			var user = Users(this.mutedUsers[0].userid);
+			if (user) {
+				user.popup("Your mute has expired.");
+				this.unmute(user);
+			} else {
+				this.mutedUsers.splice(0, 1);
+				this.runMuteTimer();
+			}
+		} else {
```

Comment:
After you have your while-loop, you don't need the `else` anymore, you can just always set

	var self = this;
	this.muteTimer = setTimeout(function () {
		self.runMuteTimer();
	}, timeUntilExpire);